client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.controller.js

```diff
@@ -0,0 +1,376 @@
+            images: this.OvhApiCloudProjectImage.Lexi().query({ serviceName: this.serviceName }).$promise
+                .then(images => {
+                    // Image types (linux, windows, ...)
+                    this.enums.imagesTypes = this.CloudImageService.constructor.getImageTypes(images);
+                    this.images = this.VirtualMachineAddService.getAugmentedImages(images);
+                })
+                .catch(this.ServiceHelper.errorHandler("cpcivm_add_step1_images_error")),
+            snapshots: this.OvhApiCloudProjectSnapshot.Lexi().query({ serviceName: this.serviceName }).$promise
+                .then(snapshots => (this.snapshots = _.map(snapshots, snapshot => _.set(snapshot, "distribution", "linux"))))
+                .catch(this.ServiceHelper.errorHandler("cpcivm_add_step1_shapshots_error")),
+            sshKeys: this.OvhApiCloudProjectSshKey.Lexi().query({ serviceName: this.serviceName }).$promise
+        })
+            .then(({ sshKeys }) => {
+                this.displayedSnapshots = _.filter(this.snapshots, { status: "active" });
+                this.displayedCustoms = [];
+                this.displayedImages = this.CloudImageService.groupImagesByType(this.images, this.enums.imagesTypes);
+                this.displayedApps = this.VirtualMachineAddService.getImageApps(this.images);
+                this.displayedSshKeys = sshKeys;
+            })
+            .catch(this.ServiceHelper.errorHandler("cpcivm_add_step1_general_error"))
+            .finally(() => {
+                this.loaders.step1 = false;
+            });
+    }
+
+    isStep1Valid () {
+        return this.model.imageType && (this.model.imageType.type !== "linux" || this.model.sshKey);
+    }
+
+    resetStep1 () {
+        _.set(this.model, "imageType", null);
+        _.set(this.model, "sshKey", null);
+        this.resetStep2();
+        this.resetAddingSshKey();
+    }
+
+    addSshKey () {
+        _.set(this.loaders, "addingSsh", true);
+        return this.OvhApiCloudProjectSshKey.Lexi().save({ serviceName: this.serviceName }, this.newSshKey).$promise
+            .then(newSshKey => {
+                this.OvhApiCloudProjectSshKey.Lexi().resetQueryCache();
+                return this.$q.all({
+                    newSshKey,
+                    sshKeys: this.OvhApiCloudProjectSshKey.Lexi().query({ serviceName: this.serviceName }).$promise
+                });
+            })
+            .then(({ newSshKey, sshKeys }) => {
+                this.displayedSshKeys = sshKeys;
+                this.model.sshKey = newSshKey;
+                this.checkSshKeyByRegion(newSshKey.regions);
+            })
+            .catch(this.ServiceHelper.errorHandler("cpcivm_add_step1_sshKey_adding_ERROR"))
+            .finally(() => {
+                this.resetAddingSshKey();
+                this.loaders.addingSsh = false;
+            });
+    }
+
+    resetAddingSshKey () {
+        _.set(this.newSshKey, "name", null);
```

Comment:
this.newSshKey.name = null.  Same for publicKey.  newSshkey is always defined so we don't need to use set.

Reply:
done